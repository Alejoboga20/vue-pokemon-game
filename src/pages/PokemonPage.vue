<template>
	<main class="game">
		<h1>Who's that pok√©mon?</h1>
		<PokemonImage />
		<PokemonOptions />
	</main>
</template>

<script lang="ts">
import PokemonImage from '@/components/PokemonImage.vue';
import PokemonOptions from '@/components/PokemonOptions.vue';

export default {
	name: 'PokemonPage',
	components: {
		PokemonImage,
		PokemonOptions,
	},
};
</script>

<style lang="css" scoped>
.game {
	font-family: 'Courier New', Courier, monospace;
	text-align: center;
}
</style>
