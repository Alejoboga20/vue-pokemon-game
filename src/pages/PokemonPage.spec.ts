import { vi } from 'vitest';
import { VueWrapper, mount, shallowMount } from '@vue/test-utils';

import { pokemonsArray } from '@/mocks/pokemons.mock';
import PokemonPage from './PokemonPage.vue';

describe('PokemonPage', () => {
	let wrapper: VueWrapper<any>;

	beforeEach(() => {
		wrapper = shallowMount(PokemonPage);
	});

	afterEach(() => {
		vi.clearAllMocks();
	});

	test('should match Snapshot after loading pokemons', () => {
		const wrapper = mount(PokemonPage, {
			data() {
				return {
					pokemonNamesArray: pokemonsArray,
					pokemon: pokemonsArray[0],
					showPokemon: false,
					showAnswer: false,
					message: '',
				};
			},
		});

		expect(wrapper.html()).toMatchSnapshot();
	});

	test('should call mixPokemonArray on mounted', () => {
		const mixPokemonArraySpy = vi.spyOn(PokemonPage.methods!, 'mixPokemonArray');
		const wrapper = shallowMount(PokemonPage);

		expect(mixPokemonArraySpy).toHaveBeenCalled();
	});
});
