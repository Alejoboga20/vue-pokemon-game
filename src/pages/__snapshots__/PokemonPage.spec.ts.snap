// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PokemonPage > should match Snapshot after loading pokemons 1`] = `
"<main data-v-2108564c=\\"\\" class=\\"game\\">
  <div data-v-2108564c=\\"\\">
    <h1 data-v-2108564c=\\"\\">Who's that pok√©mon?</h1>
    <div data-v-68dceee4=\\"\\" data-v-2108564c=\\"\\" class=\\"pokemon-container\\"><img data-v-68dceee4=\\"\\" class=\\"hidden-pokemon\\" src=\\"https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg\\" alt=\\"pokemon\\">
      <!--v-if-->
    </div>
    <ul data-v-39189cb9=\\"\\" data-v-2108564c=\\"\\">
      <li data-v-39189cb9=\\"\\">bulbasaur</li>
      <li data-v-39189cb9=\\"\\">ivysaur</li>
      <li data-v-39189cb9=\\"\\">venusaur</li>
      <li data-v-39189cb9=\\"\\">charmander</li>
    </ul>
    <!--v-if-->
  </div>
</main>"
`;
